File: springboot/src/main/java/com/edufelip/meer/security/guards/FirebaseAppCheckGuard.java
package com.edufelip.meer.security.guards;

import com.edufelip.meer.security.SecurityProperties;
import jakarta.servlet.http.HttpServletRequest;

public class FirebaseAppCheckGuard {
    public static final String APP_CHECK_HEADER = "X-Firebase-AppCheck";
    private final SecurityProperties props;

    public FirebaseAppCheckGuard(SecurityProperties props) {
        this.props = props;
    }

    public void validate(HttpServletRequest request) {
        if (!props.isRequireAppCheck() || props.isDisableAuth()) return;
        String header = request.getHeader(APP_CHECK_HEADER);
        if (header == null || header.isBlank()) {
            throw new GuardException("Missing " + APP_CHECK_HEADER + " header");
        }
        // TODO: plug real Firebase App Check validation
    }
}

File: springboot/src/main/kotlin/com/edufelip/meer/security/guards/FirebaseAppCheckGuard.kt
package com.edufelip.meer.security.guards

import com.edufelip.meer.security.SecurityProperties
import jakarta.servlet.http.HttpServletRequest

class FirebaseAppCheckGuard(private val props: SecurityProperties) {
    fun validate(request: HttpServletRequest) {
        if (!props.requireAppCheck || props.disableAuth) return
        val header = request.getHeader(APP_CHECK_HEADER)
        if (header.isNullOrBlank()) {
            throw GuardException("Missing $APP_CHECK_HEADER header")
        }
        // NOTE: Stub — real Firebase App Check token verification should be added when credentials are available.
    }

    companion object {
        const val APP_CHECK_HEADER = "X-Firebase-AppCheck"
    }
}
